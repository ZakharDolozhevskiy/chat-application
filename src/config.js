export const api = {
  host: 'http://185.13.90.140:8081/',
  channel: 'message'
};

export const routes = {
  MAIN: '/',
  SETTINGS: '/settings'
};