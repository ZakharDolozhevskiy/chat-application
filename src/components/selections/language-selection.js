import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

import Select from "@material-ui/core/es/Select/Select";
import MenuItem from "@material-ui/core/es/MenuItem/MenuItem";
import InputLabel from "@material-ui/core/es/InputLabel/InputLabel";
import FormControl from "@material-ui/core/es/FormControl/FormControl";

import { lang } from '../../config';

class LanguageSelection extends React.PureComponent {
  onChange = (event) =>
    this.props.onChange(event.target.value);

  render() {
    const { className, translate, language } = this.props;

    return (
      <FormControl className={className}>
        <InputLabel
          htmlFor="language-select"
          className="language-select-label"
        >
          {translate('Language')}:
        </InputLabel>
        <Select
          value={language}
          onChange={this.onChange}
          className="language-select"
          classes={{ icon: 'language-select-icon' }}
          inputProps={{ id: 'language-select' }}>
          <MenuItem value={lang.EN}>{translate('English')}</MenuItem>
          <MenuItem value={lang.UA}>{translate('Ukrainian')}</MenuItem>
        </Select>
      </FormControl>
    );
  }
}

LanguageSelection.propTypes = {
  onChange: PropTypes.func.isRequired,
  language: PropTypes.string.isRequired,
  translate: PropTypes.func.isRequired,
  className: PropTypes.string
};

export default styled(LanguageSelection)`
  .language-select-label {
    color: ${p => p.theme.textColor} !important;
  }
  .language-select {
    color: ${p => p.theme.textColor};

    svg {
      fill: ${p => p.theme.inputColor};
    }

    &::after, &::before { 
      border-color: ${p => p.theme.inputColor} !important;
    }
  }
`;