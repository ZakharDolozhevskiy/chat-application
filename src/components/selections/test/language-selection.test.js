import React from 'react';
import toJson from 'enzyme-to-json';
import { LanguageSelection } from '../language-selection';
import { shallow } from 'enzyme';

import { lang } from '../../../config';


describe('<LanguageSelection/>', () => {
  const callback = jest.fn();

  const component = shallow(
    <LanguageSelection
      className=""
      onChange={callback}
      translate={id => id}
      language={lang.EN}
    />
  );

  it('should render component', () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call onChange callback if changes happen', () => {
    component.instance().onChange({ target: { value: lang.UA } });
    expect(callback.mock.calls[0][0]).toBe(lang.UA);
  });
});