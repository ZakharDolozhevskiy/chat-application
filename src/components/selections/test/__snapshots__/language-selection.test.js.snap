// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LanguageSelection/> should render component 1`] = `
<WithStyles(FormControl)
  className=""
>
  <WithStyles(InputLabel)
    className="language-select-label"
    htmlFor="language-select"
  >
    Language
    :
  </WithStyles(InputLabel)>
  <WithStyles(Select)
    className="language-select"
    classes={
      Object {
        "icon": "language-select-icon",
      }
    }
    inputProps={
      Object {
        "id": "language-select",
      }
    }
    onChange={[Function]}
    value="en"
  >
    <WithStyles(MenuItem)
      value="en"
    >
      English
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      value="ua"
    >
      Ukrainian
    </WithStyles(MenuItem)>
  </WithStyles(Select)>
</WithStyles(FormControl)>
`;
